app/controllers/pieces_controller.rb

```diff
@@ -1,4 +1,7 @@
 class PiecesController < ApplicationController
+  before_action :authenticate_user!, only: [:update, :show]
+  before_action :authorize_player, only: [:show]
```

Comment:
We should authorize the player for the update action as well, no? Also, I would just remove the `only:` parameters since I think everything in this controller we need to be authenticated and authorized.